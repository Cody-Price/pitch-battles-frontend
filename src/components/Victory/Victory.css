.victory {
  background: url("../Staff/staff/staff-frame.png");
  background-size: cover;
  background-repeat: no-repeat;
  image-rendering: pixelated;
  width: 345px;
  height: 330px;
  margin: auto;
  left: 0;
  right: 0;
  position: absolute;
  overflow: hidden;
}

.victory-text {
  text-align: center;
  font-family: "chronotype";
  font-size: 3rem;
  animation: slide-in-victory 0.1s linear;
  position: absolute;
  left: 0;
  right: 0;
}

.victory p {
  text-align: center;
  font-family: "chronotype";
  width: 313px;
  padding: 1rem;
  font-size: 2rem;
  animation: slide-in-victory 0.1s linear;
  position: absolute;
  top: 50px;
  left: 0;
  right: 0;
}

@keyframes slide-in-victory {
  0% {
    left: -300px;
  }
  100% {
    left: 0;
    right: 0;
  }
}
