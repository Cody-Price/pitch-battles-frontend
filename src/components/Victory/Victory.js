import React, { Component } from "react";
import "./Victory.css";

class Victory extends Component {
  handleClick = () => {
    if (this.props.victory) {
      this.props.levelUp();
    }
  };

  render() {
    const totalTime = this.props.time.reduce((accu, num) => {
      accu += num;
      return accu;
    }, 0);

    return (
      <section onClick={this.handleClick} className="victory-screen">
        <h3 className="victory-text">VICTORY!</h3>
        {this.props.victory && (
          <p className="proceed-text">Click to proceed to next level</p>
        )}
        {this.props.time[this.props.time.length - 1] !== totalTime && (
          <p className="total-time">total time: {totalTime / 1000}</p>
        )}
        <p className="level-time">
          level time: {this.props.time[this.props.time.length - 1] / 1000}
        </p>
        {this.props.finalVictory && <p>YOU WIN!</p>}
      </section>
    );
  }
}

export default Victory;
