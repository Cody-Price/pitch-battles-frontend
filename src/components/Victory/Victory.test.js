import React from "react";
import ReactDOM from "react-dom";
import Victory from "./Victory";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("Victory", () => {
  let wrapper;
  const mockTime = [1];

  beforeEach(() => {
    wrapper = shallow(<Victory time={mockTime} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
