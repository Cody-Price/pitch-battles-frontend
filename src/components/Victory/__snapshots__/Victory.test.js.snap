// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Victory should match the snapshot 1`] = `
<section
  className="victory-screen"
  onClick={[Function]}
>
  <h3
    className="victory-text"
  >
    VICTORY!
  </h3>
  <p
    className="level-time"
  >
    level time: 
    0.001
  </p>
</section>
`;
